lib/improver/utilities/cube_checker.py

```diff
@@ -171,3 +171,49 @@ def find_dimension_coordinate_mismatch(
     else:
         mismatch = list(set(second_dim_names) - set(first_dim_names))
     return mismatch
+
```

Comment:
These functions feel like these should be properties of the Cube. 
We don't want to wrap iris methods if we don't need to.

Is it much less clearer to have:
```coord in [c.name() for c in cube.dim_coords()]```

or 
```
try:
   cube.coord_dims('coord')
except CoordinateNotFoundError:
```